### PR TITLE
Include pod exit codes and log tail in job failure errors

### LINKED ISSUE #51: Add more useful error logs when a job fails
It's difficult to tell why this job failed, would be great to get more logs and info to debug:

```
% python examples/example_gke.py
2026-02-25 05:38:23.460649: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
============================================================
Keras Remote - GKE Examples
============================================================

--- Example 1: Simple Computation (CPU) ---
Running simple_computation(10, 20) on GKE...
Packaging function and context...
Payload serialized to /tmp/tmpg8kw64fo/payload.pkl
Context packaged to /tmp/tmpg8kw64fo/context.zip
No requirements.txt found
Building container image...
Using cached container: us-docker.pkg.dev/keras-team-gcp/keras-remote/base:cpu-22d410ef9d5d
View image: https://console.cloud.google.com/artifacts/docker/keras-team-gcp/us/keras-remote/base?project=keras-team-gcp
Uploading artifacts to Cloud Storage (job: job-f6bd5a77)...
Uploaded payload to gs://keras-team-gcp-keras-remote-jobs/job-f6bd5a77/payload.pkl
Uploaded context to gs://keras-team-gcp-keras-remote-jobs/job-f6bd5a77/context.zip
View artifacts: https://console.cloud.google.com/storage/browser/keras-team-gcp-keras-remote-jobs/job-f6bd5a77?project=keras-team-gcp
Submitting job to GKEBackend...
Submitted K8s job: keras-remote-job-f6bd5a77
View job with: kubectl get job keras-remote-job-f6bd5a77 -n default
View logs with: kubectl logs -l job-name=keras-remote-job-f6bd5a77 -n default
Job keras-remote-job-f6bd5a77 running...
Pod keras-remote-job-f6bd5a77-4rbwp logs:

Deleted K8s job: keras-remote-job-f6bd5a77
Downloading result...
Traceback (most recent call last):
  File "~/jeffcarp/gh/keras-team/remote/examples/example_gke.py", line 165, in <module>
    main()
    ~~~~^^
  File "~/jeffcarp/gh/keras-team/remote/examples/example_gke.py", line 113, in main
    result = simple_computation()
  File "~/jeffcarp/gh/keras-team/remote/keras_remote/core/core.py", line 70, in wrapper
    return _execute_on_gke(
      func,
    ...<8 lines>...
      env_vars,
    )
  File "~/jeffcarp/gh/keras-team/remote/keras_remote/core/core.py", line 127, in _execute_on_gke
    return execute_remote(ctx, GKEBackend(cluster=cluster, namespace=namespace))
  File "~/jeffcarp/gh/keras-team/remote/keras_remote/backend/execution.py", line 315, in execute_remote
    raise job_error from None
  File "~/jeffcarp/gh/keras-team/remote/keras_remote/backend/execution.py", line 300, in execute_remote
    backend.wait_for_job(job, ctx)
    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
  File "~/jeffcarp/gh/keras-team/remote/keras_remote/backend/execution.py", line 134, in wait_for_job
    gke_client.wait_for_job(job, namespace=self.namespace)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/jeffcarp/gh/keras-team/remote/keras_remote/backend/gke_client.py", line 133, in wait_for_job
    raise RuntimeError(f"GKE job {job_name} failed")
RuntimeError: GKE job keras-remote-job-f6bd5a77 failed
```

### PR DESCRIPTION
**Pull request by MarcosAsh**
## Summary
- When a GKE or Pathways job fails, the RuntimeError now includes the pod's exit code, termination reason (e.g. OOMKilled), and the last 30 lines of pod logs directly in the error message.
- Previously the error was just "GKE job <name> failed" with no context on what went wrong.
- Full pod logs (100 lines) are still logged separately for deeper debugging.

Closes #51
Included below is a picture of the logs (note I used a demo script)
screenshot-2026-04-09_16-45-50

### CODE REVIEW

File: kinetic/backend/k8s_utils.py [hunk 1]
```diff
@@ -131,6 +131,64 @@ def print_pod_logs(core_v1_client, job_name, namespace):
...
+    if terminated.reason:
+      parts.append(terminated.reason)
+    if terminated.message:
+      parts.append(terminated.message.rstrip())
+    if parts:
+      return ", ".join(parts)
+  return None
+
+
+def collect_pod_failure_details(core_v1_client, job_name, namespace, tail=30):
```
Comment (JyotinderSingh) [inline kinetic/backend/k8s_utils.py]: In pathways_client.py the two `state.terminated` failure paths already include the exit code in the base message, and then `collect_pod_failure_details` appends it again via `_pod_exit_summary`

The exit code appears twice. The first failure site (`pod.status.phase == "Failed"`) does not have this problem since its base message is just `"Pathways job {job_name} failed"`. But the other two sites (current state.terminated` and `last_state.terminated`) do.

We should probably strip the exit code from the base message at those two sites (since details already contains it)

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Thanks for this contribution!

Tests look solid overall. Nice coverage on the `_pod_exit_summary` edge cases. A couple of gaps worth plugging:

* No test for the 5-pod cap. The `fetched >= 5` branch in `collect_pod_failure_details` is the most interesting boundary condition and it's untested. A test with 7 failed pods that asserts only 5 get logs + the `"... (additional failed pods omitted)"` trailer would make it better.

* No test for `ApiException` on `list_job_pods`. The try/except ApiException: return "" at the top of collect_pod_failure_details is a graceful-degradation path that should have a test confirming it returns "" instead of blowing up.

**Review by JyotinderSingh — APPROVED** [review]
LGTM. Thanks for these changes. Looks like there is a merge conflict with the main branch - could you please fix it?

Also, could you please add a screenshot of the new error logs in the PR description? This will make it easier for future reference in case that's required.

**Comment by MarcosAsh** [comment]
Just fixed it, thanks for the review! Ill add the screenshot shortly.

**Comment by MarcosAsh** [comment]
I just added a picture of the logs

**Comment by JyotinderSingh** [comment]
Awesome, thanks!
